pipelines/notion/client.py

```diff
@@ -0,0 +1,156 @@
+    ) -> str:
+        """Returns the endpoint for a given resource.
+
+        Args:
+            resource (str): The resource to get the endpoint for.
+            resource_id (str): The id of the resource.
+            subresource (str, optional): The subresource to get the endpoint for.
+
+        Returns:
+            str: The endpoint for the resource.
+        """
+        url = f'{self.BASE_URL}/{resource}/{resource_id}'
+        if subresource:
+            url += f'/{subresource}'
+        return url
+
+    def fetch_resource(
+        self, resource: str, resource_id: str, subresource: Optional[str] = None
+    ) -> Dict[str, Any]:
+        """Fetches a resource from the Notion API.
+
+        Args:
+            resource (str): The resource to fetch.
+            resource_id (str): The id of the resource.
+            subresource (str, optional): The subresource to fetch. Defaults to None.
+
+        Returns:
+            Dict[str, Any]: The resource from the Notion API.
+        """
+        url = self.get_endpoint(resource, resource_id, subresource)
+        headers = self._create_headers()
+        response = requests.get(url, headers=headers)
+        response.raise_for_status()
+        return response.json()
+
+    def send_payload(
+        self,
+        resource: str,
+        resource_id: str,
+        subresource: Optional[str] = None,
+        query_params: Optional[Dict[str, Any]] = None,
+        payload: Optional[Dict[str, Any]] = None,
+    ) -> Dict[str, Any]:
+        """Sends a payload to the Notion API using the POST method.
+
+        Args:
+            resource (str): The resource to send the payload to.
+            resource_id (str): The id of the resource.
+            subresource (str, optional): The subresource to send the payload to.
+                Defaults to None.
+            query_params (Dict[str, Any], optional): The query parameters to send
+                with the payload. Defaults to None.
+            payload (Dict[str, Any], optional): The payload to send. Defaults to None.
+
+        Returns:
+            Dict[str, Any]: The response from the Notion API.
+
+        Raises:
+            requests.HTTPError: If the response from the Notion API is not 200.
+        """
+
+        url = self.get_endpoint(resource, resource_id, subresource)
+        headers = self._create_headers()
+
+        if payload is None:
+            payload = {}
+
+        filtered_payload = self._filter_out_none_values(payload)
+
+        response = requests.post(
+            url, headers=headers, params=query_params, json=filtered_payload
+        )
+        response.raise_for_status()
+        return response.json()
+
+    def search(
+        self,
+        query: Optional[str] = None,
+        filter_criteria: Optional[Dict[str, Any]] = None,
+        sort: Optional[Dict[str, Any]] = None,
+        start_cursor: Optional[str] = None,
+        page_size: Optional[int] = None,
+    ) -> Dict[str, Any]:
```

Comment:
this is Generator so correct type would be `Iterator[...]`.

if you want your pipeline to be type-checked, place a file called `py.typed` in the `./notion` folder. this is how our Makefile selects pipelines for type checking